### PR TITLE
Refactor build

### PR DESCRIPTION
**Pull request by fir3pho3nixx**
Harmonisation of properties, before we leap to core in tests.

### CODE REVIEW

**Comment by fir3pho3nixx** [comment]
Things we need to think about: 

NUnit runs on 2 different versions for different framework monikers. Is there a common one?

I tried upgrading NUnit from 2.6.4 -> 3.6.1 for net45, Kaboom: https://github.com/fir3pho3nixx/Core/pull/6. Looks like somebody moved the cheese on pathing, cool ...  :D

How do we bring these guys(net45 && netcoreapp1.0) together in a tidy way? I think we should take the hit now. 

What do you reckon @jonorossi, how do we harmonise nunit?

**Comment by fir3pho3nixx** [comment]
Linked to #241

**Comment by jonorossi** [comment]
> I tried upgrading NUnit from 2.6.4 -> 3.6.1 for net45, Kaboom: fir3pho3nixx#6. Looks like somebody moved the cheese on pathing, cool ... :D

That seems a bit like the other current directory changes we ran into on .NET Standard. You should be able to fix not being able to find the persisted assembly in BasePEVerifyTestCase (which we can't do on .NET Core), not sure how much else is failing, the log is so big 😄 .

File: buildscripts/common.props [hunk 1]
```diff
@@ -30,11 +30,19 @@
...
 	</PropertyGroup>
 
-	<PropertyGroup Condition="'$(TargetFramework)|$(Configuration)'=='netstandard1.3|Debug'">
+  <PropertyGroup Condition="'$(TargetFramework)|$(Configuration)'=='netstandard1.3|Debug'">
+    <DefineConstants>$(DiagnosticsConstants);$(NetStandardConstants)</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)|$(Configuration)'=='netstandard1.3|Release'">
+    <DefineConstants>$(NetStandardConstants)</DefineConstants>
+  </PropertyGroup>
```
Comment (jonorossi) [inline buildscripts/common.props]: The indentation has changed from tabs to spaces here.

File: src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
   </ItemGroup>
   
   <ItemGroup>
-    <PackageReference Include="NLog" Version="4.4.5" />
+    <PackageReference Include="NLog" Version="4.4.0-beta13" />
```
Comment (jonorossi) [inline src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj]: Why are we going backwards to `4.4.0-beta13`, that one isn't even listed on nuget.org?

Ah, looks like that is what is in master, probably best for us to pick `4.4.0` for the next release then.

File: buildscripts/common.props (hunk 1 above)
Comment (fir3pho3nixx) [inline buildscripts/common.props]: Fixed this, sorry shit went south after an update. :D

File: src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj (hunk 2 above)
Comment (fir3pho3nixx) [inline src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj]: Damn! Wish this was simple. Sent you a rant email. Let's see how we go on this.

**Comment by fir3pho3nixx** [comment]
Let's jump over here for a second. #251.

File: src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj (hunk 2 above)
Comment (jonorossi) [inline src/Castle.Services.Logging.NLogIntegration/Castle.Services.Logging.NLogIntegration-VS2017.csproj]: Oh yer, forget the NLog guys did that, had to remove .NET Core support for NLog in #228 but didn't actually change project.json. We shipped with the nuspec supporting 4.4.1, can we just target .NET 4.5?

**Pull request closed without merging**